analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/ProgramState.cs

```diff
@@ -28,31 +28,37 @@ namespace SonarAnalyzer.SymbolicExecution.Roslyn
 {
     public sealed class ProgramState
     {
-        public static readonly ProgramState Empty = new(ImmutableDictionary<IOperation, SymbolicValue>.Empty, ImmutableDictionary<ISymbol, SymbolicValue>.Empty);
+        public static readonly ProgramState Empty = new();
```

Comment:
I would do a couple of changes to this approach:
- make this class a `Record`
- drop de constructor with parameters (`ProgramState`) - it's not actually needed
- update `SetOperationValue` body to `this with { OperationValue = OperationValue.SetItem(operation, value) };`
- update `SetSymbolValue` body to `this with { SymbolValue = SymbolValue.SetItem(symbol, value) };`

Reply:
That's a nice idea! I'll do the same for the `VerifierBuilder`.